language-plutus-core/src/Language/PlutusCore.hs

```diff
@@ -118,6 +119,9 @@ fileType = fmap (either prettyCfgText id . printType) . BSL.readFile
 fileTypeCfg :: Configuration -> FilePath -> IO T.Text
 fileTypeCfg cfg = fmap (either (renderCfg cfg) id . printType) . BSL.readFile
 
+checkFile :: FilePath -> IO (Maybe T.Text)
+checkFile = fmap (either (pure . prettyCfgText) id . fmap (fmap prettyCfgText . check) . parse) . BSL.readFile
```

Comment:
You might enjoy the just merged executable which will do this for you with plc typecheck :)

Reply:
Does this check the file with `check`? This is separate from typechecking.

Reply:
Ah sorry, no, I didn't see that. Possibly we should expose a `normalize` command too?